nifi-assembly/pom.xml

```diff
@@ -19,6 +19,10 @@ language governing permissions and limitations under the License. -->
     <artifactId>nifi-assembly</artifactId>
     <packaging>pom</packaging>
     <description>This is the assembly Apache NiFi</description>
+    <!--Add a blank run.as in case RPM isn't being used -->
+    <properties>
+        <nifi.run.as />
```

Comment:
I got what you are saying. 

The challenge is that the run.as user comes from a pom attribute before the RPM creation is started, meaning the the build-root used by the RPM packager already contains the correct run.as 

The sequence seems to be:

If -Prpm is enable used the line you highlighted above, if not, use https://github.com/apache/nifi/blob/master/nifi-assembly/pom.xml#L429

The only other way I imagine being possible would be using sed to replace the `run.as=` with `run.as=nifi` prior to RPM the built-root folder?